app/routes/_auth.login.tsx

```diff
@@ -1,16 +1,47 @@
 import type { LoaderArgs, ActionArgs } from '@remix-run/node';
 import { redirect } from '@remix-run/node';
 import { Form, Link, useActionData, useNavigation } from '@remix-run/react';
+import { SignInWithGoogleButton } from '~/components/ui/buttons';
 import { Card } from '~/components/ui/containers';
 import { Input } from '~/components/ui/forms';
 import { matchesHash } from '~/modules/database/crypto.server';
 import { db } from '~/modules/database/db.server';
+import { badRequest } from '~/modules/response/response.server';
+import { verifyGoogleToken } from '~/modules/session/google-auth.server';
 import { createUserSession, getUserSession } from '~/modules/session/session.server';
 
 export async function action({ request }: ActionArgs) {
   const form = await request.formData();
   const email = form.get('email');
   const password = form.get('password');
+  const credential = form.get('credential');
+
+  if (typeof credential === 'string') {
+    try {
+      const payload = await verifyGoogleToken(credential);
+      if (!payload) {
+        throw new Error('Invalid payload.');
+      }
+
+      const { name, email } = payload;
+      if (!name || !email) {
+        throw new Error('Name or email does not exist in payload.');
+      }
+
+      const user = await db.user.findUnique({ where: { email } });
+
+      if (!user) {
+        const newUser = await db.user.create({ data: { name, email } });
```

Comment:
Is this what the user wants in this case? (On the login page?) Not sure, up to discussion!

Reply:
Will had the same thoughts. I've changed the implementation so that we just notify the user that no account exists and that they should sign up instead.